### ISSUE TITLE
Early return should not take `return None` form when it is not required

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#2151]
# Rule request

## Thesis

Right now this code:

```python
    def _check_parent(self, node: ast.If) -> None:
        parent = get_parent(node)

        if parent is None:
            return None

        if isinstance(parent, _ELSE_NODES):
            body = parent.body + parent.orelse
        else:
            body = getattr(parent, 'body', [node])

        next_index_in_parent = body.index(node) + 1
        if keywords.next_node_returns_bool(body, next_index_in_parent):
            self.add_violation(SimplifiableReturningIfViolation(node))
```

passes our style check. Note `return None` part there. It is the only `return`. And it means that `None` part is not required.
Moreover, it should not be allowed`. Simple early returns must be just `return`.

**Comment by siddharth1704** [comment]
so we need just change that from `return None` to `return`

**Comment by sobolevn** [comment]
@siddharth1704 yes!

I think that we can formulate this rule as:
> If you have a single `return None`, it can only be the last item in a function's body.

**Comment by DhirajChauhan40** [comment]
@sobolevn but if we have return none in a function then it means it's meant for later use, since we don't have the value to return we do return none? please correct me if I'm wrong.

**Comment by DhirajChauhan40** [comment]
also i would like to change the code and contribute.

**Comment by DhirajChauhan40** [comment]
> @sobolevn if we have returned none in a function, it means it's meant for later use, since we don't have the value to return, we do return none? please correct me if I'm wrong.

also i would like to change the code and contribute.

**Comment by sobolevn** [comment]
We allow:

```python
def some():
   if condition:
       return None
   return other
```

We don't allow:

```python
def some():
   if condition:
       return None
   print()
```

**Comment by DhirajChauhan40** [comment]
@sobolevn can you tell me the process to contribute?
i am a beginner and i didn't understand much from contribute.md. thanks!

**Comment by DhirajChauhan40** [comment]
hey @sobolevn I understood the concept(thesis) and I did comprehend how ``` ast ```  works but I'm still not able to get how to implement 
can you provide me a sample code for some other issue that is easy to understand 
will be really helpful for a beginner like me, Thanks!

**Comment by sobolevn** [comment]
Ok, we also need to take a closer look at:

```python
def some_funct():
    if first_cond:
        return None
    if second_cond:
        return None
    print('a')
```

I would re-formulate the rule as: if all function `return` nodes are `return None` -> than we must use plain `return`.

@DhirajChauhan40 
Something like (untested):

```python
is_all_none = (
            issubclass(returning_type, ast.Return) and
            has_values and
            all(
                ret_node.value.value is None
                for ret_node in return_nodes
                if isinstance(ret_node.value, (compat.nodes.Constant, ast.NameConstant))
            )
        )
        if is_all_none:
            self.add_violation(...)
```

You can put it here: https://github.com/wemake-services/wemake-python-styleguide/blob/d41d4b9d70dcad0ece5e7ed7320560de58999fcf/wemake_python_styleguide/visitors/ast/keywords.py#L123

You would still have to:
1. Add new `violation` type in `violations/consistency.py`
2. Add tests
3. Have fun 🙂

**Comment by DhirajChauhan40** [comment]
@sobolevn 
thank you so much!!

**Comment by cdhiraj40** [comment]
@sobolevn so i was tryna run the file and I got this error in confest.py inside tests folder
```
ImportError while loading conftest '/home/thefunnyintrovert/wemake-python-styleguide/tests/conftest.py'.
  File "/home/thefunnyintrovert/wemake-python-styleguide/tests/conftest.py", line 20
    def factory(*files: str):
                      ^
SyntaxError: invalid syntax

```

**Comment by sobolevn** [comment]
Are you running pythno3.9?

**Comment by cdhiraj40** [comment]
> Are you running pythno3.9?

oh yes installing the 3.9 version solved the error but after that running test cases were giving this error
```
pytest test_consistency_return.py
ERROR: usage: pytest [options] [file_or_dir] [file_or_dir] [...]
pytest: error: unrecognized arguments: --cov=wemake_python_styleguide --cov-branch --cov-report=term-missing:skip-covered --cov-report=html --cov-report=xml --cov-fail-under=100
  inifile: /home/thefunnyintrovert/wemake-python-styleguide/setup.cfg
  rootdir: /home/thefunnyintrovert/wemake-python-styleguide

```
sorry if I'm asking a dumb doubt and sorry again for the late reply, I had studies.

**Comment by cdhiraj40** [comment]
> Are you running pythno3.9?

yes after installing the 3.9 version, it was working thanks!
sorry for the late reply i had work, i have written the test cases and now im working on the post part.

**Comment by sobolevn** [comment]
Can you please open a PR?

**Comment by cdhiraj40** [comment]
yes sure

**Comment by sathieu** [comment]
How should I write the following code without noqa?

```python
    def files_key(self, package_file: PackageFile) -> Optional[str]:
        """Get files key, to upload to. If None, uploaded as body.

        Args:
            package_file: Source package file.
        Returns:
            The files key, or None.
        """
        return  # noqa: WPS324
```

This function is overriden in child classes.

See https://gitlab.com/gitlabracadabra/gitlabracadabra/-/merge_requests/233

**Comment by sobolevn** [comment]
```python
def files_key(self, package_file: PackageFile) -> Optional[str]:  # noqa: WPS324
        """Get files key, to upload to. If None, uploaded as body.

        Args:
            package_file: Source package file.
        Returns:
            The files key, or None.
        """
```

🙂

**Comment by sathieu** [comment]
@sobolevn Now I have:
```
  157:1    DAR202 Excess "Returns" in Docstring: + return
  """Get files key, to upload to. If None, uploaded as body.

        Args:
            package_file: Source package file.

        Returns:
            The files key, or None.
        """
```

**Comment by sobolevn** [comment]
Hm, can you please open a new bug? Looks like when we have `Returns:` spec, we have a false positive.

**Comment by sathieu** [comment]
@sobolevn I created https://github.com/wemake-services/wemake-python-styleguide/issues/2323.
